earthfile2llb/converter.go

```diff
@@ -2416,6 +2428,41 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	}
 }
 
+// detectAWSCredentials determines whether AWS envs are set and/or ~/.aws
+// exists. These are then used when fetching the secrets during a build.
+func (c *Converter) detectAWSCredentials(ctx context.Context) ([]llb.RunOption, []string, error) {
+
+	credsAvail, err := awsutil.CredsAvailable()
```

Comment:
can this be moved into the into the interpreter, so we could make use of `i.errorf` in order to provide the user with the source code line that is associated with this failure?

Reply:
Updated.